src/System.Net.Http.WinHttpHandler/src/System/Net/Http/WinHttpHandler.cs

```diff
@@ -748,7 +748,11 @@ private void EnsureSessionHandleExists(WinHttpRequestState state)
                             ref optionAssuredNonBlockingTrue,
                             (uint)Marshal.SizeOf<uint>()))
                         {
-                            WinHttpException.ThrowExceptionUsingLastError();
+                            int lastError = Marshal.GetLastWin32Error();
+                            if (lastError != Interop.WinHttp.ERROR_WINHTTP_INVALID_OPTION)
```

Comment:
[Removed my last last comment]
Since we already have OS checks to detect other issues, we should ensure that we got the error on an older OS.
Alternatively, we should at least have a comment explaining why we're ignoring this particular error.